src/builtins.jl

```diff
@@ -0,0 +1,276 @@
+# This file is generated by `generate_builtins.jl`. Do not edit by hand.
+
+function getargs(args, frame)
+    nargs = length(args)-1  # skip f
+    callargs = resize!(frame.callargs, nargs)
+    for i = 1:nargs
+        callargs[i] = @eval_rhs(true, frame, args[i+1])
+    end
+    return callargs
+end
+
+function maybe_evaluate_builtin(frame, call_expr)
+    # Builtin calls can't be handled by lowering, so we have to evaluate these directly.
+    # We do this based on the call expression rather than using `to_function` because
+    # both `to_function` and `isa(f, Core.Builtin)` are quite slow, as is the (dynamic)
+    # evaluation. By having each call appearing statically in the "switch" block below,
+    # each gets call-site optimized.
+
+    args = call_expr.args
+    nargs = length(args) - 1
+    fex = args[1]
+    if isa(fex, QuoteNode)
+        f = fex.value
+    else
+        f = @eval_rhs(true, frame, fex)
+    end
+    # Builtins and intrinsics have empty method tables. We can circumvent
+    # a long "switch" check by looking for this.
+    mt = typeof(f).name.mt
+    if isa(mt, Core.MethodTable)
+        isempty(mt) || return call_expr
+    end
+    # Builtins
+    if f === <:
+        return Some{Any}(<:(@eval_rhs(true, frame, args[2]), @eval_rhs(true, frame, args[3])))
+    elseif f === ===
+        return Some{Any}(===(@eval_rhs(true, frame, args[2]), @eval_rhs(true, frame, args[3])))
+    elseif f === Core._apply
+        return Some{Any}(Core._apply(getargs(args, frame)...))
+    elseif f === Core._apply_latest
+        return Some{Any}(Core._apply_latest(getargs(args, frame)...))
+    elseif f === Core._apply_pure
+        return Some{Any}(Core._apply_pure(getargs(args, frame)...))
+    elseif f === Core._expr
+        return Some{Any}(Core._expr(getargs(args, frame)...))
+    elseif f === Core._typevar
```

Comment:
Looks like this isn't present on 1.0. One wonders if we should avoid committing `builtins.jl` and just generated it during `Pkg.build`?